### PR TITLE
first argument in apply_formatter_if now boolean, updated codebase

### PR DESCRIPTION
**Pull request by wschwab**
#382 
### What was wrong?

Order of arguments in `apply_formatter_if` needed to be flipped around to put the boolean first

### How was it fixed?

`grep`ped for instances and changed  'em

#### Cute Animal Picture

Cute animal picture

### CODE REVIEW

**Comment by carver** [comment]
I have no explanation for why github required two conflict resolutions with exactly the same details before it recognized that the PR was mergeable. I pulled c8e958e local, and it merged into master just fine.

File: web3/middleware/pythonic.py [hunk 1]
```diff
@@ -136,10 +136,10 @@ def to_hexbytes(num_bytes, val):
 
 
 LOG_ENTRY_FORMATTERS = {
-    'blockHash': apply_formatter_if(to_hexbytes(32), is_not_null),
-    'blockNumber': apply_formatter_if(to_integer_if_hex, is_not_null),
-    'transactionIndex': apply_formatter_if(to_integer_if_hex, is_not_null),
-    'transactionHash': apply_formatter_if(to_hexbytes(32), is_not_null),
+    'blockHash': apply_formatter_if(is_not_null, to_hexbytes(32)),
```
Comment (pipermerriam) [inline web3/middleware/pythonic.py]: @carver not necessarily right for this issue, but I'm thinking that we should `curry` the `apply_formatter_if` function and then define a bunch of these.

```python
apply_formatter_if_not_null = apply_formatter_if(is_not_null)
apply_formatter_if_integer = apply_formatter_if(is_integer)
```

Which cleans this up even further

```python
{
    'blockNumber': apply_formatter_if_not_null(to_integer_if_hex),
    ...
}
```

Comment (carver) [reply web3/middleware/pythonic.py]: Totally. Does the word `formatter` add much information here? It might be just as clear to use `apply_if_not_null`.

**Comment by carver** [comment]
@wschwab Sorry about all the churn, the formatting was changing pretty rapidly, but should be settled now. Would you update the PR?

If you want a hand on any of these pieces, please @ mention me here or in gitter:
1. Pull the latest master
1. In your updated master: `git diff 51b28e6bb4fe4d19760044be961cbaa75f2875cb` to take note of any new usages of `apply_formatter_if`
1. Checkout your branch
1. Rebase your commit on master
2. Flip the args of any new usages from # 2. 
2. Run `flake8 web3/ tests/` and resolve any warnings/errors.
3. Force push the change here.

**Comment by wschwab** [comment]
@carver I can't figure out what's going on at my machine. (I seem to have a little talent for creating git anomalies...) If it's okay with you, I've updated my master, and I ran through it all again. I don't have any experience with `flake8`, but I got it and ran it - 0 violations. I'll throw it into a new PR, if that's okay.

**Pull request closed without merging**